### PR TITLE
11 reconnection service create blockchain scanner service

### LINKED ISSUE #11: [reconection service] Create blockchain scanner service
Using the wallet-scanner app as a code starting point, implement a NestJS service that scans the Frequency chain for Provider Delegation events and publishes them to the reconnection-service internal queue.

### PR DESCRIPTION
**Pull request by JoeCap08055**
Initial implementation of the blockchain scanner service

Closes #11 

* Establish connection to Frequency blockchain
* Scan for events
* Publish found events to internal BullMQ

Scanner will wake up at configurable intervals to scan more blocks.

Questions to be answered:
* How many blocks to scan at each "wake-up"?
   * All available blocks?
   * Fixed number of blocks?
   * Scan for fixed amount of time?

(Current implementation scans all available blocks; will skip the next scheduled interval if a scan is still in progress)

### CODE REVIEW

File: src/interfaces/graph-update-job.interface.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { MessageSourceId, ProviderId } from '@frequency-chain/api-augment/interfaces';
+
+// Note: DSNP IDs are u64 on Frequency, but since JS 'bigint' doesn't automatically
+//       serialize to JSON, we use strings here.
```
Comment (saraswatpuneet) [inline src/interfaces/graph-update-job.interface.ts:4]: This made me realize I need to pass strings all the way to neon layer and parse, Would call a PR on that

File: env.template [hunk 2]
```diff
@@ -1,8 +1,7 @@
 # Copy this file to ".env.dev" and tweak values for local development
 FREQUENCY_URL=ws://0.0.0.0:9944
-REDIS_URL=redis://localhost
-REDIS_PORT=6379
 PROVIDER_ID=1
 PROVIDER_BASE_URL=https://some-provider
+REDIS_URL=redis://0.0.0.0:6389
```
Comment (johnmela) [inline env.template:5]: Should this line match up the ports definition in the docker-compose yaml?
`REDIS_URL=redis://0.0.0.0:6380`

File: src/interfaces/graph-update-job.interface.ts (hunk 1 above)
Comment (johnmela) [reply src/interfaces/graph-update-job.interface.ts:4]: Good to know this!

**Comment by johnmela** [comment]
Very happy to see your thoughts / comments in the codebase!

**Review by saraswatpuneet — APPROVED** [review]

**Pull request merged**